src/Servers/Kestrel/Core/src/Internal/Infrastructure/BodyControl.cs

```diff
@@ -75,6 +74,7 @@ public Stream AcceptConnect()
         {
             // CONNECT requests do not have a request or response body until after accepted,
             // unless it's a 300+ response.
+            CanHaveBody = false;
```

Comment:
What about the "unless it's a 300+ response"? Doesn't that conflict with the no body flag?

(I'm not experienced with CONNECT scenarios or code. Just raising it because of the comment)

Reply:
Will clarify the comment. `CanHaveBody` here is only for request bodies.